### DISCUSSION TITLE
Add express types on custom typescript endpoints

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#15098] [Feature Requests]
Currently the express parameters `req`, `res` and `next` are always type `any` when using with typescript.
This ends in a lot of warnings in the IDE based on the default configs. 

The problem could be resolved in installing `@types/express` as a dev dependency

https://github.com/directus/directus/blob/24b49a96620aa70ecc3610182c05865ff21b2f0f/packages/extensions-sdk/src/cli/commands/create.ts#L118-L124

**Comment by sensedrive** [comment]
The req param is also holding the _accountability_ information. Therefore the problem is not solved entirely by installing `@types/express`.

You can currently extend the Request interface by adding a typescript definition file in your source folder `request.d.ts`

```ts
declare namespace Express {
  export interface Request {
    accountability?: Accountability;
  }
}
```

**Reply by rijkvanzanten** [reply]
Wondering if there's a way to re-export the `express.d.ts` file we use in the API sourcecode: https://github.com/directus/directus/blob/main/api/src/types/express.d.ts

**Reply by northern-mindset** [reply]
A more recent discussion on this topic: https://github.com/directus/directus/discussions/20670

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
